tools/cr/post_presubmit.py

```diff
@@ -0,0 +1,159 @@
+
+            def get_severity_message():
+                if category == 'errors':
+                    # There's code never gets here as a presubmit error causes
+                    # a CI failure, so the CI may end up never launching this
+                    # script. Also, in a few tests, it has not been very clear
+                    # whether or not PRESUBMIT generates JSON entries for
+                    # errors.
+                    return (
+                        '> [!CAUTION]\n'
+                        '> You have got a presubmit error. This will cause a '
+                        'CI failure.')
+                if category == 'warnings':
+                    return (
+                        '> [!WARNING]\n'
+                        '> You have got a presubmit warning. Please address '
+                        'it if possible.')
+                if category == 'notifications':
+                    return (
+                        '> [!NOTE]\n> You have got a presubmit notification. '
+                        'No action required.')
+
+                raise ValueError(f'Unknown category: {category}')
+
+            message = report.get('message', '')
+            body_content = get_severity_message()
+            if message:
+                body_content += f'\n\n```\n{message}\n```'
+
+            items = report.get('items', [])
+            if items:
+                body_content += '\n\nItems:\n```\n'
+                for item in items:
+                    body_content += f'{item}\n'
+                body_content += '```'
+
+            body_content += f'\n\n{comment_hash}'
+
+            subprocess.run([
+                'gh', 'api',
+                f'repos/brave/brave-core/issues/{pr_number}/comments', '-X',
+                'POST', '-F', 'body=@-'
+            ],
+                           input=body_content.encode('utf-8'),
+                           check=True)
+
+    # Now we delete the old comments that the hashes didn't come up.
+    for comment in existing_comments:
+        if '<!-- presubmit-hash=' not in comment["body"]:
+            continue
+
+        comment_hash = comment["body"].split('<!-- presubmit-hash=')[1].split(
+            ' -->')[0]
+        if comment_hash not in active_report_hashes:
+            print(
+                f'Deleting comment with id {comment["id"]}, hash {comment_hash}'
+            )
+            subprocess.check_call([
+                'gh', 'api',
+                f'repos/brave/brave-core/issues/comments/{comment["id"]}',
+                '-X', 'DELETE'
+            ])
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser(description='Post-presubmit script.')
+    parser.add_argument(
+        '--input',
+        type=str,
+        default='.presubmit_results.json',
+        help='A JSON file with the output of the presubmit run (default: '
+        '.presubmit_results.json)')
+    parser.add_argument('--pr', type=str, required=True, help='The PR number.')
```

Comment:
we pass this as part of a parameter to external process. Let's make it `type=int` to minimize possible attack vector.

Reply:
Fixed.